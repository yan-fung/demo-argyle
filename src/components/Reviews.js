import "../styles/reviews.scss";

const Reviews = () => {

  return (
    <div className="review-content" data-testid="review__content">
      <div className="item">
        <div className="review-cards-container">
          <div className="title-date-container">
            <div className="title">Excellent</div>
            <div className="date">01/12/2023</div>
          </div>
          <div className="rating-name-container">
           <div className="rating" data-testid="review_rating">
             ★★★★☆
           </div>
          <div className="name">Anonymous</div>
          </div>
          <p className="body">The classes are very well prepared and organised. All the teachers are very friendly and knowledgeable. All topics are broken down into small pieces and taught step-by-step.</p>
        </div>
      </div>
      <div className="item">
        <div className="review-cards-container">
          <div className="title-date-container">
            <div className="title">Amazing platform</div>
            <div className="date">01/12/2023</div>
          </div>
          <div className="rating-name-container">
           <div className="rating" data-testid="review_rating">
             ★★★★★
           </div>
          <div className="name">Anonymous</div>
          </div>
          <p className="body">The online learning platform provides educational resources and tools for remote learning. It enables users to access courses, lectures, and educational content from anywhere with an internet connection.</p>
        </div>
      </div>
      <div className="item">
        <div className="review-cards-container">
          <div className="title-date-container">
            <div className="title">Highly recommended</div>
            <div className="date">01/12/2023</div>
          </div>
          <div className="rating-name-container">
           <div className="rating" data-testid="review_rating">
             ★★★★★
           </div>
          <div className="name">Anonymous</div>
          </div>
          <p className="body">The online learning platform provides educational resources and tools for remote learning. It enables users to access courses, lectures, and educational content from anywhere with an internet connection.</p>
        </div>
      </div>
    </div>
  );
};

export default Reviews;