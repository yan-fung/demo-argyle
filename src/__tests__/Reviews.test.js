import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Reviews from "../components/Reviews";

describe("Reviews component", () => {
  it("render correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Reviews />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
