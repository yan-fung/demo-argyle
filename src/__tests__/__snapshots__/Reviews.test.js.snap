// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component render correctly 1`] = `
<DocumentFragment>
  <div
    class="review-content"
    data-testid="review__content"
  >
    <div
      class="item"
    >
      <div
        class="review-cards-container"
      >
        <div
          class="title-date-container"
        >
          <div
            class="title"
          >
            Excellent
          </div>
          <div
            class="date"
          >
            01/12/2023
          </div>
        </div>
        <div
          class="rating-name-container"
        >
          <div
            class="rating"
            data-testid="review_rating"
          >
            ★★★★☆
          </div>
          <div
            class="name"
          >
            Anonymous
          </div>
        </div>
        <p
          class="body"
        >
          The classes are very well prepared and organised. All the teachers are very friendly and knowledgeable. All topics are broken down into small pieces and taught step-by-step.
        </p>
      </div>
    </div>
    <div
      class="item"
    >
      <div
        class="review-cards-container"
      >
        <div
          class="title-date-container"
        >
          <div
            class="title"
          >
            Amazing platform
          </div>
          <div
            class="date"
          >
            01/12/2023
          </div>
        </div>
        <div
          class="rating-name-container"
        >
          <div
            class="rating"
            data-testid="review_rating"
          >
            ★★★★★
          </div>
          <div
            class="name"
          >
            Anonymous
          </div>
        </div>
        <p
          class="body"
        >
          The online learning platform provides educational resources and tools for remote learning. It enables users to access courses, lectures, and educational content from anywhere with an internet connection.
        </p>
      </div>
    </div>
    <div
      class="item"
    >
      <div
        class="review-cards-container"
      >
        <div
          class="title-date-container"
        >
          <div
            class="title"
          >
            Highly recommended
          </div>
          <div
            class="date"
          >
            01/12/2023
          </div>
        </div>
        <div
          class="rating-name-container"
        >
          <div
            class="rating"
            data-testid="review_rating"
          >
            ★★★★★
          </div>
          <div
            class="name"
          >
            Anonymous
          </div>
        </div>
        <p
          class="body"
        >
          The online learning platform provides educational resources and tools for remote learning. It enables users to access courses, lectures, and educational content from anywhere with an internet connection.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
